### PR TITLE
Improved YML and JSON file loading

### PR DESCRIPTION
**Pull request by MosheEichler**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7186

## Description
Changed function from yml.load to get_yaml
Changed function from json.load to get_json

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/tests/update_rn_test.py [hunk 1]
```diff
@@ -2100,10 +2100,8 @@ def test_update_docker_image_in_yml(self, mocker):
             "longRunning: false\n"
         )
 
-        with open(
-            "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
-        ) as file:
-            pack_data = json.load(file)
+        json_file = "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
+        pack_data = get_json(json_file)
```
Comment (kobymeir) [inline demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:2104]:
```suggestion
        pack_data = get_json("demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json")
```

File: demisto_sdk/commands/update_release_notes/tests/update_rn_test.py [hunk 2]
```diff
@@ -2155,10 +2153,8 @@ def test_update_docker_image_in_yml_when_RN_aleady_exists(self, mocker):
         """
         from demisto_sdk.commands.update_release_notes.update_rn import UpdateRN
 
-        with open(
-            "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
-        ) as file:
-            pack_data = json.load(file)
+        json_file = "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
```
Comment (kobymeir) [inline demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:2156]: same here, simplify it

File: demisto_sdk/commands/update_release_notes/tests/update_rn_test.py [hunk 3]
```diff
@@ -2228,10 +2224,8 @@ def test_add_and_modify_files_without_update_docker_image(self, mocker):
 
         from demisto_sdk.commands.update_release_notes.update_rn import UpdateRN
 
-        with open(
-            "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
-        ) as file:
-            pack_data = json.load(file)
+        json_file = "demisto_sdk/commands/update_release_notes/tests_data/Packs/Test/pack_metadata.json"
```
Comment (kobymeir) [inline demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:2227]: same here, simplify it

File: demisto_sdk/commands/update_release_notes/tests/update_rn_test.py [hunk 4]
```diff
@@ -2285,10 +2279,8 @@ def test_new_integration_docker_not_updated(self, mocker):
...
```
Comment (kobymeir) [inline demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:2282]: same here, simplify it

File: demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py [hunk 5]
```diff
@@ -35,6 +35,10 @@ def temp_dir():
         rmtree(temp)
 
 
+def json_file(config_file):
```
Comment (kobymeir) [inline demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py:38]: what's the just of this function?

File: demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py [hunk 6]
```diff
@@ -287,8 +286,8 @@ def test_add_custom_pack_with_missing_args(
                 assert str_in_call_args_list(logger_info.call_args_list, err)
 
             try:
-                with open(f"{tmp_output_dir}/{expected_path}") as config_file:
-                    config_file_info = json.load(config_file)
+                json_file = f"{tmp_output_dir}/{expected_path}"
+                config_file_info = get_json(json_file)
```
Comment (kobymeir) [inline demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py:290]:
```suggestion
                config_file_info = get_json(f"{tmp_output_dir}/{expected_path}")
```

**Review by kobymeir — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/generate_unit_tests/tests/test_files/outputs/malwarebazaar_all.py [hunk 7]
```diff
@@ -2,12 +2,14 @@
...
+
+from demisto_sdk.commands.common.tools import get_json
+
 SERVER_URL = 'https://test_url.com'
 
 
 def util_load_json(path):
-    with io.open(path, mode='r', encoding='utf-8') as f:
-        return json.loads(f.read())
+    return get_json(path)
```
[the remark this answers is not in the document]
Comment (dorschw) [reply demisto_sdk/commands/generate_unit_tests/tests/test_files/outputs/malwarebazaar_all.py:12]: this is a test file, it's not part of the SDK, no need to replace it in these files at all

File: demisto_sdk/commands/update_xsoar_config_file/update_xsoar_config_file.py [hunk 8]
```diff
@@ -142,14 +142,7 @@ def update_xsoar_config_data(self, section_name, data_to_update):
...
     def get_xsoar_config_data(self):
-        config_file_info = {}
-        if os.path.exists(self.file_path):
-            with open(self.file_path) as config_file:
-                try:
-                    config_file_info = json.load(config_file)
-                except json.JSONDecodeError:  # In case that the file exits but empty
-                    pass
-        return config_file_info
+        return get_json(self.file_path)
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply demisto_sdk/commands/update_xsoar_config_file/update_xsoar_config_file.py:145]: returns empty dict

File: demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py (hunk 5 above)
Comment (MosheEichler) [reply demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py:38]: I don't know probably added by mistake....

**Comment by kobymeir** [comment]
please fix the pre-commit failure before i review, it's a large PR

**Pull request closed without merging**